core/src/test/java/org/infinispan/protostream/ProtobufUtilTest.java

```diff
@@ -224,6 +225,28 @@ public void testJsonWithDifferentFieldOrder() throws Exception {
       assertEquals(createAccount(), account);
    }
 
+   @Test
+   public void vectorFieldsTranscoding() throws Exception {
+      SerializationContext context = createContext();
+      Item.ItemSchema.INSTANCE.registerSchema(context);
+      Item.ItemSchema.INSTANCE.registerMarshallers(context);
+
+      Item java = new Item("c7", new byte[]{7, 7, 7}, new float[]{1.1f, 1.1f, 1.1f}, "bla bla bla");
+      byte[] proto = ProtobufUtil.toWrappedByteArray(context, java);
+      String json = ProtobufUtil.toCanonicalJSON(context, proto);
+
+      // Covert back JSON -> Proto -> Java
+      byte[] proto2 = ProtobufUtil.fromCanonicalJSON(context, new StringReader(json));
+      Item java2 = ProtobufUtil.fromWrappedByteArray(context, proto2);
+      assertEquals(java, java2);
+
+      // Moreover, we want to parse also Java byte[] / Proto bytes fields in the JSON format `[7,7,7]`
+      String alternativeJson = "{\"_type\":\"Item\",\"code\":\"c7\",\"byteVector\":[7,7,7],\"floatVector\":[1.1,1.1,1.1],\"buggy\":\"bla bla bla\"}";
+      byte[] proto3 = ProtobufUtil.fromCanonicalJSON(context, new StringReader(alternativeJson));
+      Item java3 = ProtobufUtil.fromWrappedByteArray(context, proto3);
+      assertEquals(java, java3);
+   }
+
```

Comment:
Is there a test around to verify that the mapping between json array of int -> java int[] still works?
I can't find it

Reply:
good idea. I added a int[] field to cover the case